docs/environment-variables.md

```diff
@@ -0,0 +1,36 @@
+# Environment Variables
+
+This document outlines the set of environment variables that can be used to customize the behaviours at different levels.
+These environment variables are typically added to test out experimental features and should not be needed by most users.
+Note that these environment variables may be removed at any time.
+
+## Controller
+
+| Name | Type | Description|
+|----------|------|------------|
+| `ALL_POD_CHANGES_SIGNIFICANT` | `bool` |  Whether to consider all pod changes as significant during pod reconciliation. |
+| `ALWAYS_OFFLOAD_NODE_STATUS` | `bool` | Whether to always offload the node status. |
+| `ARCHIVED_WORKFLOW_GC_PERIOD` | `time.Duration` | The periodicity for GC of archived workflows. |
+| `ARGO_TRACE` | `bool` | Whether to enable tracing statements in Argo components. |
+| `DEFAULT_REQUEUE_TIME` | `time.Duration` | The requeue time for the rate limiter of the workflow queue. |
+| `LEADER_ELECTION_IDENTITY` | `string` | The ID used for workflow controllers to elect a leader. |
+| `MAX_OPERATION_TIME` | `time.Duration` | The maximum time a workflow operation is allowed to run for before requeuing the workflow onto the work queue. |
+| `OFFLOAD_NODE_STATUS_TTL` | `time.Duration` | The TTL to delete the offloaded node status. Currently only used for testing. |
+| `POD_GC_TRANSIENT_ERROR_MAX_REQUEUES` | `int` | The maximum number of requeues of pods in the GC queue when encountering transient errors. |
```

Comment:
Actually, can you remove this and add it in https://github.com/argoproj/argo-workflows/pull/5060 so that it doesn't fail the check when I merge https://github.com/argoproj/argo-workflows/pull/5091?

Reply:
Removed and rebased.

Reply:
I'll add this in #5060 once this has been merged.